tests/topotests/multicast-pim-bsm-topo1/test_mcast_pim_bsmp_01.py

```diff
@@ -579,6 +579,7 @@ def test_BSR_higher_prefer_ip_p0(request):
     write_test_footer(tc_name)
 
 
+@pytest.mark.skip()
```

Comment:
nope, this is not acceptable to just mark a test as skip.  We must investigate and fix the issue whether it is in (a) the test or (b) the new proposed code.

Reply:
@kuldeepkash could you please check this test? (`test_BSR_CRP_with_blackhole_address_p1`)
I believe the test might be broken and my fix for blackhole routes reveals this.

Reply:
Hi @idryzhov , 

We have looked into the execution logs, and could see that blackhole route created for RP is getting accepted but without your fix routes was getting dropped as expected.

This is causing this test failure. Is it possible to get dumps for "show ip route and show ip pim nexthop" from node l1 and f1.

Reply:
Hi @idryzhov and @donaldsharp ,  I have enhanced test and tested it with your changes and in master branch. Raised PR: https://github.com/FRRouting/frr/pull/8157